docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -1982,6 +1990,560 @@ Emits a {TimelockUpdated} event.
+
+Returns the computed identifier of a transaction containing a single call.
+
+[.contract-item]
+[[MultisigComponent-hash_transaction_batch]]
+==== `[.contract-item-name]#++hash_transaction_batch++#++(self: @ContractState, calls: Span<Call>, salt: felt252)++` [.item-kind]#external#
+
+Returns the computed identifier of a transaction containing a batch of calls.
+
+[#MultisigComponent-Internal-Functions]
+==== Internal functions
+
+[.contract-item]
+[[MultisigComponent-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, quorum: u32, signers: Span<ContractAddress>)++` [.item-kind]#internal#
+
+Initializes the Multisig component with the initial `quorum` and `signers`.
+
+Emits a `SignerAdded` event for each signer added.  
+Emits a `QuorumUpdated` event if the quorum changes.
+
+[.contract-item]
+[[MultisigComponent-resolve_tx_state]]
+==== `[.contract-item-name]#++resolve_tx_state++#++(self: @ContractState, id: TransactionID) → TransactionState++` [.item-kind]#internal#
+
+Resolves and returns the current state of the transaction with the given `id`.
+
+[.contract-item]
+[[MultisigComponent-assert_one_of_signers]]
+==== `[.contract-item-name]#++assert_one_of_signers++#++(self: @ContractState, caller: ContractAddress)++` [.item-kind]#internal#
+
+Asserts that the `caller` is one of the registered signers.
+
+[.contract-item]
+[[MultisigComponent-assert_tx_exists]]
+==== `[.contract-item-name]#++assert_tx_exists++#++(self: @ContractState, id: TransactionID)++` [.item-kind]#internal#
+
+Asserts that a transaction with the given `id` exists.
+
+[.contract-item]
+[[MultisigComponent-assert_only_self]]
+==== `[.contract-item-name]#++assert_only_self++#++(self: @ContractState)++` [.item-kind]#internal#
+
+Asserts that the caller is the contract itself.
+
+[.contract-item]
+[[MultisigComponent-_add_signers]]
+==== `[.contract-item-name]#++_add_signers++#++(ref self: ContractState, new_quorum: u32, signers_to_add: Span<ContractAddress>)++` [.item-kind]#internal#
+
+Adds new signers and updates the quorum.
+
+[.contract-item]
+[[MultisigComponent-_remove_signers]]
+==== `[.contract-item-name]#++_remove_signers++#++(ref self: ContractState, new_quorum: u32, signers_to_remove: Span<ContractAddress>)++` [.item-kind]#internal#
+
+Removes existing signers and updates the quorum.
+
+[.contract-item]
+[[MultisigComponent-_replace_signer]]
+==== `[.contract-item-name]#++_replace_signer++#++(ref self: ContractState, signer_to_remove: ContractAddress, signer_to_add: ContractAddress)++` [.item-kind]#internal#
+
+Replaces an existing signer with a new signer.
+
+[.contract-item]
+[[MultisigComponent-_change_quorum]]
+==== `[.contract-item-name]#++_change_quorum++#++(ref self: ContractState, new_quorum: u32)++` [.item-kind]#internal#
+
+Updates the quorum value to `new_quorum` if it differs from the current quorum.
+
```

Comment:
Any reason not to include the emitted events and requirements as they're written in the in-code doc?